app/component/map/map.cjsx

```diff
@@ -66,13 +57,16 @@ class Map extends React.Component
       @setBounds(newProps)
 
   componentWillUnmount: ->
-    @context.getStore('PositionStore').removeChangeListener @onChange
+    @context.getStore('PositionStore').removeChangeListener @onPositionChange
     @context.getStore('EndpointStore').removeChangeListener @onChange
 
-  onChange: =>
+  onChange: (endPointChange) =>
+    if endPointChange in ['set-origin']
+      origin = @context.getStore('EndpointStore').getOrigin()
+      @refs.map.getLeafletElement().setView([origin.lat, origin.lon])
```

Comment:
Uh, this is not a very React-way of doing things, they should be passed in the props for the map component in the render function